data-raw/bucharest.R

```diff
@@ -1,62 +1,18 @@
-library(CRiSp)
-library(osmdata)
-library(dplyr)
-library(sf)
-
+# Set the parameters
 city_name <- "Bucharest"
 river_name <- "Dâmbovița"
 epsg_code <- 32635
 bbox_buffer <- 2000
 
-city_boundary <- get_osm_city_boundary(city_name) |>
-  st_transform(epsg_code)
-st_crs(city_boundary)$wkt <- gsub("°|º", "\\\u00b0",
-                                  st_crs(city_boundary)$wkt)
-
-bb <- getbb(city_name)
-aoi <- define_aoi(bb, epsg_code, bbox_buffer)
-bbox_expanded <- aoi |> st_transform(4326) |> st_bbox()
-
-river_centerline <- osmdata_as_sf("waterway", "river", bb)$osm_multilines |>
-  filter(name == river_name) |>
-  st_transform(epsg_code) |>
-  st_geometry() |>
-  st_intersection(st_buffer(aoi, bbox_buffer))
-st_crs(river_centerline)$wkt <- gsub("°|º", "\\\u00b0",
-                                     st_crs(river_centerline)$wkt)
-
-river_surface <- osmdata_as_sf("natural", "water", bb)
-river_surface <- river_surface$osm_multipolygons |>
-  bind_rows(river_surface$osm_polygons) |>
-  st_transform(epsg_code) |>
-  st_filter(river_centerline, .predicate = st_intersects) |>
-  st_geometry() |>
-  st_union()
-st_crs(river_surface)$wkt <- gsub("°|º", "\\\u00b0",
-                                  st_crs(river_surface)$wkt)
-
-highway_values <- c("motorway", "primary", "secondary", "tertiary")
-streets <- osmdata_as_sf("highway", highway_values, bb)
-streets <- merge_streets(streets) |>
-  select("highway") |>
-  st_transform(epsg_code)
-st_crs(streets)$wkt <- gsub("°|º", "\\\u00b0", st_crs(streets)$wkt)
-
-railways <- osmdata_as_sf("railway", "rail", bbox_expanded)
-railways_lines <- railways$osm_lines |>
-  select("railway") |>  # only keep "railway" column
-  rename(type = `railway`) |>  # rename it to "type"
-  st_transform(epsg_code)
-st_crs(railways_lines)$wkt <- gsub("°|º", "\\\u00b0",
-                                   st_crs(railways_lines)$wkt)
+# Fetch the data
+bucharest <- CRiSp::get_osmdata(city_name, river_name,
+                                crs = epsg_code, buffer = bbox_buffer)
 
-bucharest <- list(
-  bb = bb,
-  boundary = city_boundary,
-  river_centerline = river_centerline,
-  river_surface = river_surface,
-  streets = streets,
-  railways_lines = railways_lines
-)
+# Fix encoding issue in the WKT string of city boundary
+lapply(names(bucharest), function(layer) {
+  sf::st_crs(bucharest[[layer]])$wkt <- gsub("°|º", "\\\u00b0",
+                                             sf::st_crs(bucharest[[layer]])$wkt)
+})
```

Comment:
Uhm, I had to modify this in the following way in order to get it to work:
```suggestion
fix_wkt_encoding <- function(x) {
  wkt <- sf::st_crs(x)$wkt
  sf::st_crs(x)$wkt <- gsub("°|º", "\\\u00b0", wkt)
  x
}
bucharest <- lapply(bucharest, fix_wkt_encoding)
```

I think the .rda file should also be updated